Automattic/jetpack
# VideoPress: Add delete action to video details page

Fixes #27163

## Proposed changes:

* Extracts the `DeleteVideoConfirmationModal` component
* Adds the `VideoDetailsActions` component to the video details page
* Adds the `Add to new post` and `Delete video` actions
* Adds loading state to the video details page
* Fixes a console warning

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the VideoPress dashboard
* Delete a video via the quick actions. It should work as normal
* Go the the video details page of any video
* Click on the "more actions" button (1)
![2023-02-28_12-50-23](https://user-images.githubusercontent.com/8486249/221905953-f0809052-dfbf-44a8-aec1-70382ec6893a.png)
* Click on `Add to new post` (2)
* Check that a new tab/window opens with the selected video in a new post
* Go back to the dashboard video details page
* Click on `Delete video` (3)
* Check that a confirmation modal opens
* Confirm the deletion
* Check that the page goes on a visual loading state
* Check that the video is deleted successfully

https://user-images.githubusercontent.com/8486249/221907163-4fe28f8e-982b-43d1-bc8f-632b617df413.mp4
